### PR TITLE
fix(upgrade): handle correctly topology database update

### PR DESCRIPTION
**Pull request by adr-mo**
## Description

This PR intends to fix an issue regarding the database update of topology table

🏷️ Fixes **MON-23830**

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x
- [x] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Do 23.04.x -> 23.04.8 update
Do 23.04.x -> 23.10.x update

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by dmyios — APPROVED** [review]

File: centreon/www/install/php/Update-23.04.1.php [hunk 1]
```diff
@@ -26,21 +26,53 @@
...
+};
+
 $removeNagiosPathImg = function(CentreonDB $pearDB) {
     $selectStatement = $pearDB->query("SELECT 1 FROM options WHERE `key`='nagios_path_img'");
     if($selectStatement->rowCount() > 0) {
         $pearDB->query("DELETE FROM options WHERE `key`='nagios_path_img'");
     }
 };
 
+$alterTopologyForTopologyUrlSubstitue = function(CentreonDB $pearDB) {
```
Comment (adr-mo) [inline centreon/www/install/php/Update-23.04.1.php:48]:
```suggestion
$alterTopologyForTopologyUrlSubstitue = function(CentreonDB $pearDB): void {
```

**Review by adr-mo — COMMENTED** [review]
Auto review

**Review by jeremyjaouen — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

File: centreon/www/install/php/Update-23.04.8.php [hunk 2]
```diff
@@ -17,4 +17,50 @@
...
+
+//error specific content
+$versionOfTheUpgrade = 'UPGRADE - 23.04.8: ';
+$errorMessage = '';
+
+$alterTopologyForFeatureFlag = function(CentreonDB $pearDB): void {
+    if (!$pearDB->isColumnExist('topology', 'topology_feature_flag')) {
+        $pearDB->query(
+            <<<'SQL'
+                ALTER TABLE `topology`
```
Comment (callapa) [inline centreon/www/install/php/Update-23.04.8.php:33]: This SQL command `ALTER TABLE` cause an implicit commit after executing.

File: centreon/www/install/php/Update-23.04.1.php [hunk 3]
```diff
@@ -26,21 +26,53 @@
...
     if($selectStatement->rowCount() > 0) {
         $pearDB->query("DELETE FROM options WHERE `key`='nagios_path_img'");
     }
 };
 
+$alterTopologyForTopologyUrlSubstitue = function(CentreonDB $pearDB): void {
+    if(!$pearDB->isColumnExist('topology', 'topology_url_substitute')) {
+        $pearDB->query(
+            <<<'SQL'
+                ALTER TABLE `topology`
```
Comment (callapa) [inline centreon/www/install/php/Update-23.04.1.php:52]: This SQL command ALTER TABLE cause an implicit commit after executing.

File: centreon/www/install/php/Update-23.04.1.php [hunk 4]
```diff
@@ -26,21 +26,53 @@
 $versionOfTheUpgrade = 'UPGRADE - 23.04.1: ';
 $errorMessage = '';
 
+$alterTopologyForFeatureFlag = function(CentreonDB $pearDB): void {
+    if (!$pearDB->isColumnExist('topology', 'topology_feature_flag')) {
+        $pearDB->query(
+            <<<'SQL'
+                ALTER TABLE `topology`
```
Comment (callapa) [inline centreon/www/install/php/Update-23.04.1.php:33]: This SQL command ALTER TABLE cause an implicit commit after executing.

**Review by callapa — APPROVED** [review]

**Pull request merged**